modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GoClientCodegen.java

```diff
@@ -382,13 +382,21 @@ public String toOperationId(String operationId) {
         // if the return type is not primitive, import encoding/json
         for (CodegenOperation operation : operations) {
             if(operation.returnBaseType != null && needToImport(operation.returnBaseType)) {
-                Map<String, String> customImport = new HashMap<String, String>();
-                customImport.put("import", "encoding/json");
-                imports.add(customImport);
+                imports.add(createMapping("import", "encoding/json"));
                 break; //just need to import once
             }
         }
 
+        // this will only import "strings" "fmt" if there are items in pathParams
+        for (CodegenOperation operation : operations) {
+            if(operation.pathParams != null && operation.pathParams.size() > 0) {
+                imports.add(createMapping("import", "fmt"));
```

Comment:
The `postProcessOperations` and `postProcessModels` are pretty useful in meeting language-specified requirement.